### PR TITLE
Replaced Renderer hide_custom_tags options to be a  Hash instead of Array

### PR DESCRIPTION
**Pull request by frederico**
Replaced Renderer `hide_custom_tags` options to be a `Hash` instead of `Array`.
  There are are edge cases which we want to replace the content on hide custom tags.
  All documentations can be followed on `Renderer` and `HideCustomTag` classes.

### CODE REVIEW

File: lib/lm_docstache/parser.rb [hunk 1]
```diff
@@ -128,23 +136,20 @@ def replace_variables_in_document!
...
-
         text_node.content = text
       end
     end
 
     def has_skippable_variable?(text)
-      return true if hide_custom_tags.find { |pattern| text =~ pattern }
+      return true if hide_custom_tags.find { |(pattern, value)| text =~ pattern }
       !!special_variable_replacements.find do |(pattern, value)|
-        pattern = pattern.is_a?(String) ? /{{#{pattern}}}/ : /{{#{pattern.source}}}/
```
Comment (leandro) [inline lib/lm_docstache/parser.rb:139]: Why did you remove this line here?

File: lib/lm_docstache/parser.rb [hunk 2]
```diff
@@ -128,23 +136,20 @@ def replace_variables_in_document!
...
         text =~ pattern && value == false
       end
     end
 
     def special_variable_replacement(text)
       Array(
         special_variable_replacements.find do |(pattern, value)|
-          pattern = pattern.is_a?(String) ? /{{#{pattern}}}/ : /{{#{pattern.source}}}/
```
Comment (leandro) [inline lib/lm_docstache/parser.rb:147]: Same question here.

File: lib/lm_docstache/version.rb [hunk 3]
```diff
@@ -1,3 +1,3 @@
 module LMDocstache
-  VERSION = "2.1.2"
```
Comment (leandro) [inline lib/lm_docstache/version.rb:2]: We don't need to change the major version...do we? Are we having breaking changes?

File: spec/hide_custom_tags_spec.rb [hunk 4]
```diff
@@ -46,26 +52,37 @@
...
-            expect(run_node.children.first.name).to eq('rPr')
-          end
+          next unless run_node.text =~ regexp_tag
+          expect(run_node.at_css('w|rPr w|color').first[1]).to eq('FFFFFF')
+          expect(run_node.children.first.name).to eq('rPr')
         end
       end
     end
-    context 'giving a document without rpr' do
+    context 'giving a document without rpr and blcok tags on the left' do
```
Comment (leandro) [inline spec/hide_custom_tags_spec.rb]:
```suggestion
    context 'giving a document without rpr and block tags on the left' do
```

**Review by leandro — COMMENTED** [review]
I have left some comments.

File: lib/lm_docstache/parser.rb (hunk 1 above)
Comment (frederico) [reply lib/lm_docstache/parser.rb:139]: Because we have a method that makes this happens? https://github.com/boost-legal/lm-docstache/pull/11/files#diff-473de67f52e8aca33647ef59951c5e059ee4ed9ab5fdf9287a121332db52fbbbR51

File: lib/lm_docstache/parser.rb (hunk 2 above)
Comment (frederico) [reply lib/lm_docstache/parser.rb:147]: Same response here.

File: lib/lm_docstache/version.rb (hunk 3 above)
Comment (frederico) [reply lib/lm_docstache/version.rb:2]: Yes, it's a breaking change. We replaced Array to Hash in the options and might break if someone updates to minor version.
